### ISSUE TITLE
Could you please check the code?

### ISSUE THREAD

**Issue by Dong1P** [draimundo/LAN#1]
Hello, I am a student studying your code and paper.

First of all, it is an honor to send an e-mail to you as a member of the Radu group.
When I run this code, I get the result that the PSNR does not exceed 14.5dB.
I read your paper and tried the same parameters. But, it didn't work.
May I know if you have ever studied using this code?
Could you please check the code? I can't find any errors no matter how I look at it.

thank you

**Comment by draimundo** [comment]
Hi @Dong1P, what dataset are you using? 

A PSNR this low generally means the network is producing random outputs/isn't trained right. I used this code to train the network, so there shouldn't be any problems.

For troubleshooting, you could try producing output images with your network. Checking what the pairs you are feeding to the network really look like may also help you. Also, starting with a simple loss function (e.g MSE only) may help isolate the problem. Finally, note that a pre-trained model is provided.

Best,
draimundo

**Comment by Dong1P** [comment]
Thank you for quick response.

I downloaded your code again and tried running "evaluate_model.py".

My Tensorflow version is 2.5.

For test data, 2400 raw images of data were randomly selected from AIM 2022 Mobile ISP train data.

Pre-trained network used "LAN/models/LAN_iteration_200000.ckpt.index" 

I also checked the test result images. But the correct image was not generated.

The baseline code (https://github.com/MediaTek-NeuroPilot/mai21-learned-smartphone-isp) works fine. 
But this code doesn't work. I can't find a problem.

There is no normalization part in the code. Could it be a problem?
Raw image read uses the following sentences:
"In = np.asarray(imageio.imread(val_directory_phone + str(i) + '.png'))"

Here is the result:


python evaluate_model.py

2022-07-12 13:28:15.034345: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0
The following parameters will be applied for testing:
Path to the dataset: raw_images/
Path to VGG-19 network: vgg_pretrained/imagenet-vgg-verydeep-19.mat
Path to RGB data from DSLR: fujifilm/
Path to Raw data from phone: mediatek_raw/
Path to Raw-to-RGB model network: models/
Restore iteration:200000
Batch size: 10
Loading testing data...
100%|████████████████████████████████████████████████████████████████████████████████████████| 2416/2416 [00:15<00:00, 160.45it/s]
Testing data was loaded

Losses - iter: 200000-> loss_psnr: 11.8792; loss_mse: 0.0790832; loss_l1: 0.241712; loss_ssim: 0.468114; loss_ms_ssim: 0.53798; loss_content: 11.1199; loss_lpips: 0.712999; loss_huber: 0.0395416; loss_tv: 0;

**Comment by draimundo** [comment]
You were right @Dong1P , there was a mistake in the code. I cleaned up my code to publish it here, removing some modularity (while still keeping a lot of it, I hope it can help you for further studies!) but put the wrong activation functions in the model definition, which didn't match the paper or the pre-trained model. 

I updated the code and got the right PSNR on the test set. Please check my last commit 1d934dedb410392862b182e70453cfc32f17d880  and let me know if it fixed the issue.

**Comment by draimundo** [comment]
I guess no news is good news @Dong1P , feel free to comment again if something doesn't work
